InteractiveHtmlBom/core/ibom.py

```diff
@@ -66,6 +66,10 @@ def skip_component(m, config):
     if config.blacklist_virtual and m.attr == 'Virtual':
         return True
 
+    # skip components if dedicated kicad dnp flag is set
+    if m.extra_fields and m.extra_fields.get('kicad_dnp'):
+        return True
+
```

Comment:
Your workflow is not everyone's workflow. But your change forces everyone to make components dnp in ibom if they are dnp in kicad. You already are exposing it as a normal extra field, just let users use already existing dnp configuration that ibom supports.

Reply:
Due to the KiCad 7 update this will get common. The dnp field was more like a workaround. So which field the user should choose when there is no?

tbh I'am not a python developer. It took me several hours to understand a bit of your code and to change something that works. Maybe someone who knows what all needs to be changed should implement this feature the right way then.

Reply:
Your change converts kicad's dnp flag into "kicad_dnp" field. All I'm saying is remove these lines to not force users to use this new "kicad_dnp" field as dnp and allow them to opt into it.